<hr>
<center>
   <font size='4'> <B> Tarefa: </B> Realizar amostragem estatrificada de bernoulli com base de dados enviada pelo sigaa  </font> <br>
    <br>
    <font size='3'> <I> <B> Feito por: </B> Yasmin de Miranda Nobre </I> </font>
</center> 
<hr>


# Código Inicial

In [2]:
library("readxl")
library("TeachingSampling")
dados = read_xlsx("Dados_Estratificado.xlsx")
pis = c(0.4,0.3,0.2,0.4)
U1 = dados[dados$Estrato==1,] #população 1
U2 = dados[dados$Estrato==2,] #população 2
U3= dados[dados$Estrato==3,] #população 3
U4 = dados[dados$Estrato==4,] #população 4
n1= nrow(U1)
n2 = nrow(U2)
n3=nrow(U3)
n4=nrow(U4)
N = nrow(dados) #quantidade de observações
tpi = NULL

In [3]:
#visualizando tamanhos populacionais
N
n1
n2
n3
n4

[1] 227

[1] 29

[1] 60

[1] 98

[1] 40

In [6]:
#visualizando totais populacionais de y
sum(U1$yk)
sum(U2$yk)
sum(U3$yk)
sum(U4$yk)

[1] 874

[1] 10271

[1] 6752

[1] 10898

# Primeira amostra (a_s1) com primeira população (U1)

In [7]:
set.seed(1612)

s1 = S.BE(n1,pis[1])

s1 = s1[s1!=0]; s1 #ver individuos selecionados

a_s1 = dados[s1,] #amostra selecionada

a_s1

E.BE(a_s1$yk,pis[1]) #estimativa feita na amostra s1

tpi = append(tpi,1137.5) #adicionar valor de tpi ao vetor com todos os tpi

[1]  3  4  6  7 10 11 14 15 16 17 20 21 26 28 29

Estrato k  yk xk      
1  1        3 33 64.75044
2  1        4 26 51.93935
3  1        6 22 41.98452
4  1        7 26 51.09727
5  1       10 40 79.71031
6  1       11 34 65.08357
7  1       14 31 59.95311
8  1       15 30 57.44968
9  1       16 30 59.13758
10 1       17 35 69.55108
11 1       20 23 45.91595
12 1       21 24 45.58220
13 1       26 32 61.15906
14 1       28 35 69.11069
15 1       29 34 67.92532

N    y         
Estimation     37.5 1137.50000
Standard Error  7.5  230.57266
CVE            20.0   20.27012
DEFF            Inf   35.25364

<B> Comentários: </B> <br>

Cve do pi-estimador está abaixo de 25%, ou seja, podemos confiar na estimativa do total da variavel y como 1.137,5, em média, e inferir que y varia pouco. O erro padrão é 230.57, o que, aparentemente, é uma variação pequena para essa população <br>
<br>
Quanto à estimativa do total de observações populacional, a estimativa é de aproximadamente 38, com erro padrão de aproximadamente 8 unidades. Sendo assim, essa estimativa está perto da real que é 29 

# Segunda amostra (a_s2) com segunda população (U2)

In [77]:
set.seed(1612)

s2 = S.BE(n2,pis[2])

s2 = s2[s2!=0]; s2 #ver individuos selecionados

a_s2 = dados[s2,]

E.BE(a_s2$yk,pis[2]) 

tpi = append(tpi,7176.7) #adicionar valor de tpi ao vetor com todos os tpi

[1]  3  6  7 10 11 14 15 16 17 20 21 26 29 34 35 36 37 38 39 40 47 48 60

N        y          
Estimation     76.66667 7176.666667
Standard Error 13.37494 1584.927969
CVE            17.44557   22.084458
DEFF                Inf    2.544058

<B> Comentários: </B> <br>

Cve do pi-estimador também está abaixo de 25% ,indicando que a estimativa do total de y varia pouco em relação a 7.176. Ele é um pouco maior do que o anterior. Embora o valor real não seja contemplado no intervalo estimado quando consideramos o erro padrão, na prática, ele está perto da estimativa. Os dados indicam que a população U2 possui, de modo geral, mais da variavel y que a população U1 <br>
<br>
Quanto à estimativa do total de observações populacional, a estimativa é de aproximadamente 77, perto da real: 60

# Terceira amostra (a_s3) com terceira população (U3)

In [78]:
set.seed(1612)

s3 = S.BE(n3,pis[3])

s3 = s3[s3!=0]; s3 #ver individuos selecionados

a_s3 = dados[s3,]

E.BE(a_s3$yk,pis[3]) 

tpi = append(tpi,13925) #adicionar valor de tpi ao vetor com todos os tpi

[1]  3  6  7 10 15 16 17 26 34 35 36 37 40 47 60 65 75 77 78 79 84 85 98

N         y          
Estimation     115.00000 13925.00000
Standard Error  21.44761  2987.53745
CVE             18.65010    21.45449
DEFF                 Inf     3.91470

<B> Comentários: </B> <br>

Assim como nos casos anteriores, o cve do pi-estimador é baixo, sugerindo homogeneidade nos dados de y. Seu cve é o menor que o da amostra s2 mas maior que da amsotra s1. Entretanto, o valor real é de seis mil e oitocentos, aproximadamente, quase a metade do valor estimado. <br>
<br>
Quanto à estimativa do total de observações populacional, a estimativa é de aproximadamente 115. Quando consideramos o erro padrão, vemos que esse valor é preciso.

# Quarta amostra (a_s4) com quarta população (U4)

In [79]:
set.seed(1612)

s4 = S.BE(n4,pis[4])

s4 = s4[s4!=0]; s4 #ver individuos selecionados

a_s4 = dados[s4,]

E.BE(a_s4$yk,pis[4]) 

tpi = append(tpi,4275) #adicionar valor de tpi ao vetor com todos os tpi

[1]  3  4  6  7 10 11 14 15 16 17 20 21 26 28 29 34 35 36 37 38 39 40

N         y          
Estimation     55.000000 4275.000000
Standard Error  9.082951  948.568659
CVE            16.514456   22.188741
DEFF                 Inf    2.139952

<B> Comentários: </B> <br>

Cve do pi-estimador também indica homogeneidade no valor de y e é o maior se comparado aos anteriores. Em relação ao valor estimado do total de y, é o que possui o segundo menor valor, estando atras do valor estimado em a_s1. Entretanto, o valor real é de, em média,10.898, enquanto o estimado é de 4.275<br>

O DEFF em todos as amostragens que fizemos indica que a variancia estimada por meio do método bernoulli é maior do que a que obteriamos caso fizessemos pelo meio usual, de maneira aleatoria e simples.<br>

De modo geral, percebemos que na primeira população o valor total estimado de y é muito baixo comparado ao valor na segunda, terceira e quarta (mil, 7 mil, 14 mil e 4 mil, respectivamente). Entretanto, como a amostragem foi feita pelo metodo estratificado, sabemos que a estimativa do total da amostra s (s = união de a_s1,a_s2,a_s3 e a_s4) não será influenciada por esse valor discrepante e irá refletir, de maneira significante, o total estimado de y na população U (U = união dos individuos presentes em U1,U2,U3 e U4) <br>
<br>
Quanto à estimativa do total de observações populacional, a estimativa é de 55 quando a real é 40. Levando em conta o erro padrão, o valor de N não está entre o intervalo estimado. Entretanto, a variabilidade nesse valor não é alta, como indica o cve de 16% 

# Cálculo do pi-estimador de s e variancia estimada do mesmo

In [82]:
tpi

[1]  1137.5  7176.7 13925.0  4275.0

In [83]:
tpih = sum(tpi); tpih
var_s = c((230.5727)^2,(1584.9280)^2,(2987.5375^2),(948.5687)^2)
var_sh = sum(var_s);var_sh
cve = sqrt(var_sh)/tpih;cve*100

[1] 26514.2

[1] 12390323

[1] 13.27586

In [84]:
mean(tpi)

[1] 6628.55

<B> Comentários: </B> <br>

O total estimado de y na população é de 27 mil, aproximadamente, algo que faz sentido considerando que a o valor total de y na terceira população é de quase 14 mil. <br>

O cve do pi-estimador é de apenas 13%, indicando que podemos confiar na estimativa acima, visto que há pouca variabilidade no valor do estimador.<br>